Content.IntegrationTests/Tests/Power/PowerTest.cs

```diff
@@ -712,6 +714,101 @@ public async Task TestFullBatteryEfficiencyDemandPassThrough()
             await pairTracker.CleanReturnAsync();
         }
 
+        /// <summary>
+        ///     Checks that if there is insufficient supply to meet demand, generators will run at full power instead of
+        ///     having generators and batteries sharing the load. 
+        /// </summary>
+        [Test]
+        public async Task TestSupplyPrioritized()
+        {
+            await using var pairTracker = await PoolManager.GetServerClient(new PoolSettings { NoClient = true, ExtraPrototypes = Prototypes });
+            var server = pairTracker.Pair.Server;
+            var mapManager = server.ResolveDependency<IMapManager>();
+            var entityManager = server.ResolveDependency<IEntityManager>();
+            var gameTiming = server.ResolveDependency<IGameTiming>();
+            PowerConsumerComponent consumer = default!;
+            PowerSupplierComponent supplier1 = default!;
+            PowerSupplierComponent supplier2 = default!;
+            PowerNetworkBatteryComponent netBattery1 = default!;
+            PowerNetworkBatteryComponent netBattery2 = default!;
+            BatteryComponent battery1 = default!;
+            BatteryComponent battery2 = default!;
+
+            await server.WaitAssertion(() =>
+            {
+                var map = mapManager.CreateMap();
+                var grid = mapManager.CreateGrid(map);
+
+                // Layout is two generators, two batteries, and one load. As to why two: because previously this test
+                // would fail ONLY if there were more than two batteries present, because each of them tries to supply
+                // the unmet load, leading to a double-battery supply attempt and ramping down of power generation from
+                // supplies.
+
+                // Actual layout is Battery Supply, Load, Supply,  Battery
+
+                // Place cables
+                for (var i = -2; i <= 2; i++)
+                {
+                    grid.SetTile(new Vector2i(0, i), new Tile(1));
```

Comment:
Gross.
Make a map for this, maybe adding a simple system for tagged test entities in the meantime so the big bunch of spawns below can be simplified to `_testTagSystem.GetEntity("Battery1")` and so on.

Reply:
In general: yeah I 100% agree, there needs to be a better way of testing setups that require properly setting up entities on a grid/map, but I'd also rather do that in some separate PR and not unnecessarily overcomplicate this one

Reply:
> In general: yeah I 100% agree, there needs to be a better way of testing setups that require properly setting up entities on a grid/map, but I'd also rather do that in some separate PR and not unnecessarily overcomplicate this one

Fairnuf.